src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -121,16 +121,16 @@ BoundExpression MakeDeclarationPattern(SyntaxNode syntax, BoundExpression input,
                 if (input.Type == type)
                 {
                     // CONSIDER: this can be done whenever input.Type is a subtype of type for improved code
-                    var assignment = _factory.AssignmentExpression(_factory.Local(target), input);
+                    var assignment = _factory.AssignmentExpression(target, input);
                     var result = requiresNullTest
-                        ? _factory.ObjectNotEqual(_factory.Local(target), _factory.Null(type))
+                        ? _factory.ObjectNotEqual(target, _factory.Null(type))
```

Comment:
It is documented in BoundNodes.xml. The fact that lowering is correct is verified by our compiler unit tests.

---
In reply to: [77407349](https://github.com/dotnet/roslyn/pull/13535#discussion_r77407349) [](ancestors = 77407349,77397610,77388658,77387889,77387439)